subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/HierarchicalFileWatcherUpdater.java

```diff
@@ -92,6 +94,11 @@ public void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarch
         updateWatchedHierarchies(root);
     }
 
+    @Override
+    public SnapshotHierarchy buildStarted(SnapshotHierarchy root) {
+        return movedHierarchyHandler.handleMovedHierarchies(root);
```

Comment:
Don't we need to update the watchers after invalidating parts of the VFS? I'd at least expect that we tell the watcher updater that we are not watching the moved path any more.

I think a good way would be to call `updateWatchedHierarchies` if the new root is different.

Reply:
Yes, good point.